### PR TITLE
Update arborx

### PR DESCRIPTION
**Pull request by Rombur**
This PR works (I've only tried CPU) if I revert https://github.com/arborx/ArborX/pull/222 @masterleinad can you take a look at what we need to make `sendAcrossNetwork` work. One problem is obviously [this](https://github.com/arborx/ArborX/pull/222#pullrequestreview-431922791) but when I fixed it. I got the wrong results for the `MeshFree` tests.

### CODE REVIEW

**Comment by Rombur** [comment]
This is ready for review. The travis fails because of doxygen in arborx.

File: packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp [hunk 1]
```diff
@@ -123,44 +125,51 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DetailsDistributedSearchTreeImpl,
...
+    //        Kokkos::RangePolicy<ExecutionSpace>( 0, comm_size * comm_rank ),
+    //        KOKKOS_LAMBDA( int i ) {
+    //            for ( int j = 0; j < DIM; ++j )
+    //                v_ref( i, j ) = j * comm_rank;
+    //        } );
+    //    Kokkos::fence();
+    //
+    //    Helper<DeviceType>::checkSendAcrossNetwork( comm, ranks_v, v_exp,
+    //    v_ref,
+    //                                                success, out );
```
Comment (masterleinad) [inline packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp:172]: Why did you comment all of this?

File: packages/Meshfree/test/tstMovingLeastSquaresOperatorSimpleProblem.cpp [hunk 2]
```diff
@@ -113,38 +117,12 @@ void checkResults( std::vector<double> const &values,
 TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( MovingLeastSquaresOperatorSimpleProblem,
                                    corner_cases, DeviceType )
 {
-    // No target point and no source point.
-    {
-        std::vector<std::array<double, DIM>> source_points = {};
```
Comment (masterleinad) [inline packages/Meshfree/test/tstMovingLeastSquaresOperatorSimpleProblem.cpp:118]: Why did you delete this?

File: packages/Meshfree/test/tstNearestNeighborOperator.cpp [hunk 3]
```diff
@@ -112,20 +107,8 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( NearestNeighborOperator, unique_source_point,
     DataTransferKit::NearestNeighborOperator<DeviceType> nnop(
         comm, source_points, target_points );
 
-    Kokkos::View<double *, DeviceType> source_values( "in" );
-    Kokkos::View<double *, DeviceType> target_values( "out" );
-
-    // violate pre condition of apply (source not properly sized)
```
Comment (masterleinad) [inline packages/Meshfree/test/tstNearestNeighborOperator.cpp:118]: ... and this

File: packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp (hunk 1 above)
Comment (Rombur) [reply packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp:172]: That's when I was debugging my problem with `sendAcrossNetwork`. It should not be commented.

File: packages/Meshfree/test/tstMovingLeastSquaresOperatorSimpleProblem.cpp (hunk 2 above)
Comment (Rombur) [reply packages/Meshfree/test/tstMovingLeastSquaresOperatorSimpleProblem.cpp:118]: For two of the cases the code does not throw and the last one it throws inside Kokkos instead of DTK. Since the code still throws (from Kokkos) when you try call `apply`, I didn't really try to understand why it stopped throwing during the construction.

File: packages/Meshfree/test/tstNearestNeighborOperator.cpp (hunk 3 above)
Comment (Rombur) [reply packages/Meshfree/test/tstNearestNeighborOperator.cpp:118]: Same reason as before: we don't throw a DTK exception anymore and I don't really care about it. The user just needs to be careful.

**Comment by dalg24** [comment]
Can you comment to our relaxing the tolerance from 1e-14 to 1e-6

**Comment by masterleinad** [comment]
> The travis fails because of doxygen in arborx.

We should probably just disable doxygen parsing arborx (either in this pull request or another one).

**Comment by Rombur** [comment]
> Can you comment to our relaxing the tolerance from 1e-14 to 1e-6

Yeah, that's because we went from `double` to `float` in the point coordinates. I _think_ that the important computations are done on double but we need to cast from `float` to `double` at some point. I looked at a few checks were I've relaxed the tolerance and it was clear that it was just due to a different round-off (value of 2.500001 instead of 2.5).

**Comment by dalg24** [comment]
> > Can you comment to our relaxing the tolerance from 1e-14 to 1e-6
> 
> Yeah, that's because we went from `double` to `float` in the point coordinates. I _think_ that the important computations are done on double but we need to cast from `float` to `double` at some point. I looked at a few checks were I've relaxed the tolerance and it was clear that it was just due to a different round-off (value of 2.500001 instead of 2.5).

Do you reckon we might have an issue here?  We could recompute the distance in dtk in principle.

**Comment by Rombur** [comment]
> > > Can you comment to our relaxing the tolerance from 1e-14 to 1e-6
> > 
> > 
> > Yeah, that's because we went from `double` to `float` in the point coordinates. I _think_ that the important computations are done on double but we need to cast from `float` to `double` at some point. I looked at a few checks were I've relaxed the tolerance and it was clear that it was just due to a different round-off (value of 2.500001 instead of 2.5).
> 
> Do you reckon we might have an issue here? We could recompute the distance in dtk in principle.

In the current version of the code, this wouldn't help since **all** the points in dtk are using float now. I'll see how much work it is to use float only in arborx and double everywhere else.

**Comment by Rombur** [comment]
OK I've switched back to double and it works using a tolerance of `1e-14` instead of `1e-6`. We get narrowing warning but that can be fixed in another PR

**Comment by dalg24** [comment]
> OK I've switched back to double and it works using a tolerance of `1e-14` instead of `1e-6`. We get narrowing warning but that can be fixed in another PR

Great.  Tanks for fixing this.
